pkg/util/ring.go

```diff
@@ -31,3 +31,25 @@ func IsInReplicationSet(r *ring.Ring, ringKey uint32, address string) (bool, err
 	}
 	return false, nil
 }
+
+// IsAssignedKey replies wether the given component instance address is in the ReplicationSet responsible for the given key or not.
+//
+// The result will be defined based on the tokens assigned to each ring component.
+func IsAssignedKey(ringClient ring.ReadRing, instance HasDedicatedAddress, key string) bool {
+	bufDescs, bufHosts, bufZones := ring.MakeBuffersForGet()
+	token := TokenFor(key, "" /* labels */)
+	rs, err := ringClient.Get(token, ring.WriteNoExtend, bufDescs, bufHosts, bufZones)
+	if err == nil {
+		addr := instance.GetInstanceAddr()
+		if !rs.Includes(addr) {
+			return false
+		}
+	}
+
+	return true
+}
+
+// HasDedicatedAddress specify an instance support for replying its own address.
+type HasDedicatedAddress interface {
```

Comment:
Naming is hard :)
It doesn't sound correct to me. What do you think about using `RingInstanceInfoReader` or simply going back to previous implementation to pass the address directly?

Reply:
Hahaha it is!

Agreed, it looks odd. I returned to pass the address directly, lmk what you think.